sample-apps/go-sample-app/collection/client.go

```
@@ -59,25 +54,37 @@ var traceCommonLabels = []attribute.KeyValue{
 	attribute.String("port", cfg.Port),
 }
 
-// StartClient starts the OTEL controller which periodically collects signals and exports them.
+// StartClient starts the traces and metrics providers which periodically collects signals and exports them.
 // Trace exporter and Metric exporter are both configured.
 func StartClient(ctx context.Context) (func(context.Context) error, error) {
 
+	if id, present := os.LookupEnv("INSTANCE_ID"); present {
+		testingId = "_" + id
+	}
+	res := resource.NewWithAttributes(
+		semconv.SchemaURL,
+		semconv.ServiceName("go-sample-app"),
+	)
+	if _, present := os.LookupEnv("OTEL_RESOURCE_ATTRIBUTES"); present {
+		res, _ = resource.New(ctx, resource.WithFromEnv())
```

Don't ignore this error.

---

Error is now handled here